lib/std/bit_set.zig

```diff
@@ -110,6 +110,31 @@ pub fn IntegerBitSet(comptime size: u16) type {
             self.mask |= maskBit(index);
         }
 
+        /// Changes the value of all bits in the specified range to
+        /// match the passed boolean.
+        pub fn setRangeValue(self: *Self, range: Range, value: bool) void {
+            assert(range.end <= bit_length);
+            assert(range.start <= range.end);
+            if (range.start == range.end) return;
+            if (MaskInt == u0) return;
+
+            const start_bit = @intCast(ShiftInt, range.start);
+
+            var mask = std.math.boolMask(MaskInt, true) << start_bit;
+            if (range.end != bit_length) {
+                const end_bit = @intCast(ShiftInt, range.end);
+                mask &= std.math.boolMask(MaskInt, true) >> @truncate(ShiftInt, @as(usize, @bitSizeOf(MaskInt)) - @as(usize, end_bit));
```

Comment:
This `@truncate(@as(usize) - @as(usize))` hack would be nice to avoid, but I couldn't find a way. If I use the `(@bitSizeOf(MaskInt) - 1) - (end_bit - 1)` method as in the other sets, the compiler complains that `1` isn't a valid `u0`, because it doesn't understand the early-return when `MaskInt` is a `u0`.